src/sdl2_compat.c

```diff
@@ -9072,6 +9055,22 @@ SDL_GetPrefPath(const char *org, const char *app)
     return retval ? SDL3_strdup(retval) : NULL;
 }
 
+SDL_DECLSPEC SDL_Locale *SDL_GetPreferredLocales(void)
+{
+    SDL_Locale *retval = NULL;
+    int i, count;
+    const SDL_Locale * const *locales = SDL3_GetPreferredLocales(&count);
+    if (locales) {
+        retval = (SDL_Locale *)SDL3_calloc(count + 1, sizeof(*retval));
```

Comment:
Why doesn't this one just claim the temporary memory?

Reply:
Because in SDL2 this function returned an array of structures, not an array of pointers.